lib/importers/tpas_satisfaction/retriever.rb

```diff
@@ -0,0 +1,21 @@
+require 'mail_retriever'
+
+module Importers
+  module TpasSatisfaction
+    class Retriever
+      def initialize(config:, mail_retriever: MailRetriever)
+        @config = config
+        @mail_retriever = mail_retriever
+      end
+
+      def process_emails
+        retriever = @mail_retriever.new(config: @config)
+        emails = retriever.search(search_keys: @config.search_string, file_name_regexp: @config.file_name_regexp)
+
+        succeeded, failed = emails.partition { |email| yield email }
+        failed.each { |email| Rails.logger.warn("TPAS email failed to process: #{email.subject}") }
```

Comment:
I can't remember what you did elsewhere, but perhaps for some consistency and future decoupling we should inject the `Rails.logger`?